Add some more hints to the 3.0 upgrade and move data migration down (#7921)
sulu/sulu: UPGRADE-3.x.md

                prefix: 'optional/path/prefix'
```

</details>

<details>
  <summary>Sulu Azure Config to Flysystem Config</summary>

Before:

```yaml
sulu_media:
    storage: google_cloud
    storages:
        google_cloud:
            key_file_path: '/path/to/key.json'
            bucket_name: 'sulu-bucket'
            path_prefix: 'optional path prefix'
```

New:

```yaml
flysystem:
    storages:
        default.storage:
            adapter: 'azure'
            options:
                client: 'azure_client_service' # The service ID of the MicrosoftAzure\Storage\Blob\BlobRestProxy instance
                container: 'container_name'
                prefix: 'optional/path/prefix'
```

</details>

If you want use a [different storage](https://github.com/thephpleague/flysystem-bundle/blob/3.x/docs/2-cloud-storage-providers.md) for Sulu you can configure it here:

```yaml
# config/packages/sulu_media.yaml
sulu_media:
    storage:
        flysystem_service: 'default.storage' # this is default and not required to be configured.
```

This will only create the service `sulu_media.storage` as the alias to `sulu_media.storage.*` services has been removed.

### Preview Services changed

> Most of the PreviewBundle services are now internal and only the new `PreviewDefaultsProviderInterface` is
> now the way to communicate with the Bundle.

Reviewer: ```suggestion
Most of the services in the PreviewBundle are now internal. The new `PreviewDefaultsProviderInterface` is
now the primary means of interacting with the bundle.
```
